services/src/main/java/org/keycloak/services/clientpolicy/executor/ScopeClientRegistrationEnforceExecutor.java

```diff
@@ -0,0 +1,53 @@
+package org.keycloak.services.clientpolicy.executor;
+
+import org.jboss.logging.Logger;
+import org.keycloak.component.ComponentModel;
+import org.keycloak.events.Errors;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.services.clientpolicy.ClientPolicyContext;
+import org.keycloak.services.clientpolicy.ClientPolicyException;
+import org.keycloak.services.clientpolicy.ClientUpdateContext;
+
+public class ScopeClientRegistrationEnforceExecutor implements ClientPolicyExecutorProvider {
+
+    private static final Logger logger = Logger.getLogger(ScopeClientRegistrationEnforceExecutor.class);
+
+    private final KeycloakSession session;
+    private final ComponentModel componentModel;
+
+    public ScopeClientRegistrationEnforceExecutor(KeycloakSession session, ComponentModel componentModel) {
+        this.session = session;
+        this.componentModel = componentModel;
+    }
+
+    @Override
+    public String getName() {
+        return componentModel.getName();
+    }
+
+    @Override
+    public String getProviderId() {
+        return componentModel.getProviderId();
+    }
+
+    @Override
+    public void executeOnEvent(ClientPolicyContext context) throws ClientPolicyException {
+        switch (context.getEvent()) {
+            case REGISTERED:
+                ClientUpdateContext registeredClientContext = (ClientUpdateContext) context;
+                registeredClientContext.getRegisteredClient().setFullScopeAllowed(false);
+                break;
+            case UPDATE:
```

Comment:
It seems that it lacks the following logic that `ScopeClientRegistrationPolicy` implements.
```
        if (context.getClient().isFullScopeAllowed() == null) {
            return;
        }
```
How about the following?
```
                ClientUpdateContext updateClientContext = (ClientUpdateContext) context;
                if (updateClientContext.getProposedClientRepresentation().isFullScopeAllowed() == null) {
                    return;
                }
```